projects/pgai/tests/vectorizer/test_vectorizer_cli.py

```diff
@@ -0,0 +1,420 @@
+def cli_postgres_container():
+    """Provides a clean Postgres container for CLI testing"""
+    with PostgresContainer("timescale/timescaledb-ha:pg16", driver=None) as postgres:
+        yield postgres
+
+
+@pytest.fixture
+def cli_db(
+    cli_postgres_container: PostgresContainer,
+) -> Generator[dict[str, Any], None, None]:
+    """Creates a test database with pgai installed"""
+    role = "tsdbquerier"
+    password = "my-password"
+    db_host = cli_postgres_container._docker.host()  # type: ignore
+
+    # Connect and setup initial database
+    with psycopg.connect(
+        cli_postgres_container.get_connection_url(host=db_host),
+        autocommit=True,
+    ) as conn:
+        # Setup test user
+        conn.execute(sql.SQL("DROP USER IF EXISTS {}").format(sql.Identifier(role)))
+        conn.execute(
+            sql.SQL("CREATE USER {} WITH SUPERUSER PASSWORD {}").format(
+                sql.Identifier(role), sql.Literal(password)
+            )
+        )
+
+        # Install pgai
+        conn.execute("CREATE EXTENSION IF NOT EXISTS ai CASCADE")
+
+        yield {
+            "container": cli_postgres_container,
+            "conn": conn,
+            "event_db_config": {
+                "host": db_host,
+                "port": int(
+                    cli_postgres_container.get_exposed_port(cli_postgres_container.port)
+                ),
+                "db_name": cli_postgres_container.dbname,
+                "ssl_mode": "disable",
+                "role": role,
+                "password": password,
+            },
+        }
+
+
+@pytest.fixture
+def cli_db_url(cli_db: dict[str, Any]) -> str:
+    """Constructs database URL from the cli_db fixture"""
+    config = cli_db["event_db_config"]
+    return f"postgres://{config['role']}:{config['password']}@{config['host']}:{config['port']}/{config['db_name']}?sslmode={config['ssl_mode']}"
+
+
+def test_worker_no_tasks(cli_db_url: str):
+    """Test that worker handles no tasks gracefully"""
+    result = CliRunner().invoke(vectorizer_worker, ["--db-url", cli_db_url, "--once"])
+
+    # It exits successfully
+    assert result.exit_code == 0
+    assert "no vectorizers found" in result.output.lower()
+
+
+@pytest.fixture
+def configured_vectorizer_and_source_table(
+    cli_db: dict[str, Any],
+    test_params: tuple[int, int, int, str, str],
+) -> int:
+    """Creates and configures a vectorizer for testing"""
+    num_items, concurrency, batch_size, chunking, formatting = test_params
+
+    with cli_db["conn"].cursor(row_factory=dict_row) as cur:
+        # Cleanup from previous runs
+        cur.execute("SELECT id FROM ai.vectorizer")
+        for row in cur.fetchall():
+            cur.execute("SELECT ai.drop_vectorizer(%s)", (row["id"],))
+
+        # Drop tables if they exist
+        cur.execute("DROP VIEW IF EXISTS blog_embedding")
+        cur.execute("DROP TABLE IF EXISTS blog_embedding_store")
+        cur.execute("DROP TABLE IF EXISTS blog")
```

Comment:
I wish we didn't have to manually clean up. One idea would be to create a new database for each test, so we know that we're dealing with a fresh db. We can do it as a small follow-up PR though.